qpc/insights/upload.py

```diff
@@ -147,16 +168,54 @@ def _validate_args(self):
                      self.report_id)))
         print(_(messages.INSIGHTS_RETRIEVE_REPORT % self.report_id))
 
+    def verify_report_details(self):
+        """
+        Verify that the report contents are a valid deployments report.
+
+        :returns boolean regarding report validity
+        """
+        # validate required keys
+        report_contents = extract_json_from_tarfile(self.tmp_tar_name)
```

Comment:
I think that you can validate from memory before the tar file is created.

Reply:
Great catch! Fixed